test/forge/Strategies.sol

```diff
@@ -0,0 +1,1829 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { Test } from "forge-std/Test.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { ITransparentUpgradeableProxy } from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import { TestFixture } from "./TestFixture.sol";
+
+import { Order, Strategy, Strategies } from "../../contracts/carbon/Strategies.sol";
+
+import { AccessDenied, ZeroValue, InvalidAddress, InvalidFee, InvalidIndices } from "../../contracts/utility/Utils.sol";
+import { PPM_RESOLUTION } from "../../contracts/utility/Constants.sol";
+
+import { CarbonController } from "../../contracts/carbon/CarbonController.sol";
+import { Strategies } from "../../contracts/carbon/Strategies.sol";
+import { Pairs } from "../../contracts/carbon/Pairs.sol";
+import { TestVoucher } from "../../contracts/helpers/TestVoucher.sol";
+import { TestCarbonController } from "../../contracts/helpers/TestCarbonController.sol";
+import { TestERC20FeeOnTransfer } from "../../contracts/helpers/TestERC20FeeOnTransfer.sol";
+
+import { IVoucher } from "../../contracts/voucher/interfaces/IVoucher.sol";
+
+import { Token, NATIVE_TOKEN } from "../../contracts/token/Token.sol";
+
+contract StrategiesTest is TestFixture {
+    using Address for address payable;
+
+    // strategy update reasons
+    uint8 private constant STRATEGY_UPDATE_REASON_EDIT = 0;
+    uint8 private constant STRATEGY_UPDATE_REASON_TRADE = 1;
+
+    uint32 private constant DEFAULT_TRADING_FEE_PPM = 2000;
+    uint32 private constant NEW_TRADING_FEE_PPM = 300_000;
+
+    uint256 private constant FETCH_AMOUNT = 5;
+
+    /**
+     * @dev triggered when the network fee is updated
+     */
+    event TradingFeePPMUpdated(uint32 prevFeePPM, uint32 newFeePPM);
+
+    /**
+     * @dev triggered when a strategy is created
+     */
+    event StrategyCreated(
+        uint256 id,
+        address indexed owner,
+        Token indexed token0,
+        Token indexed token1,
+        Order order0,
+        Order order1
+    );
+
+    /**
+     * @dev triggered when a strategy is deleted
+     */
+    event StrategyDeleted(
+        uint256 id,
+        address indexed owner,
+        Token indexed token0,
+        Token indexed token1,
+        Order order0,
+        Order order1
+    );
+
+    /**
+     * @dev triggered when a strategy is updated
+     */
+    event StrategyUpdated(
+        uint256 indexed id,
+        Token indexed token0,
+        Token indexed token1,
+        Order order0,
+        Order order1,
+        uint8 reason
+    );
+
+    /**
+     * @dev Emitted when `tokenId` token is transferred from `from` to `to`.
+     */
+    event Transfer(address indexed from, address indexed to, uint256 indexed tokenId);
```

Comment:
Can we import all of these events instead of redeclaring them?

Reply:
Only in solidity 0.8.21